modules/authentication.py

```diff
@@ -185,24 +186,24 @@ def verify_response(
         # Macaroon not valid anymore, needs refresh
         if verified['account'] is None:
             empty_session(session)
-            return {
-                'status_code': 307,
-                'redirect': url_login,
-                'reason': 'Need login'
-            }
+            raise ApiVerifyResponseError(
+                'Need login',
+                307,
+                url_login
+            )
 
         # Not authorized content
         if response.status_code == 401 and not verified['allowed']:
-            return {
-                'status_code': 401,
-                'redirect': url_account,
-                'reason': 'Not authorized'
-            }
+            raise ApiVerifyResponseError(
+                'Not authorized',
+                401,
+                url_account
+            )
 
         # The package doesn't exist
         if verified['account'] is not None and response.status_code == 404:
-            return {
-                'status_code': 404,
-                'redirect': url_account,
-                'reason': 'Not found'
-            }
+            raise ApiVerifyResponseError(
```

Comment:
Actually all we're basically doing in this function, if you remove the macaroon refresh stuff, is pass request errors back to the client. Maybe it's not even worth wrapping these - we should simply do `response.raise_for_status` and let the client catch them? Or do you think there's an argument for wrapping it in our own error type?

If we do wrap it in our own error type, couldn't it be generic? E.g.:

if response.status_code >= 400:
    raise ApiResponseError(
        "The API returned a bad response",
        response=response
    )